core/controllers/feedback_test.py

```diff
@@ -554,3 +556,216 @@ def test_feedback_threads_with_suggestions(self):
         }
         self.assertDictContainsSubset(
             expected_suggestion_dict, response['suggestion'])
+
+    def test_get_feedback_threads_with_no_text_and_no_updated_status_raises_400(
+            self):
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_html_response('/create/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        feedback_services.create_thread(
+            feconf.ENTITY_TYPE_EXPLORATION, self.EXP_ID, self.owner_id_1,
+            'a subject', 'some text', has_suggestion=True)
+
+        response_dict = self.get_json(
+            '%s/%s' % (feconf.FEEDBACK_THREADLIST_URL_PREFIX, self.EXP_ID))
+        threadlist = response_dict['suggestion_thread_dicts']
+        thread_id = threadlist[0]['thread_id']
+
+        new_content = state_domain.SubtitledHtml(
+            'content', 'new content html').to_dict()
+        change = {
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+            'state_name': 'Welcome!',
+            'new_value': new_content
+        }
+        suggestion_models.GeneralSuggestionModel.create(
+            suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION, self.EXP_ID,
+            1, suggestion_models.STATUS_IN_REVIEW, self.owner_id_1,
+            None, change, 'score category', thread_id)
```

Comment:
You have not "put" the model. I think once you create it you need to call .put() (or .save(), I can't seem to remember) on the model explicitly to save it...

More importantly, why do you need the suggestion?

Reply:
The create function does the put so no need to put again.

I need the suggestion as the controller requires a suggestion to work. Look at ThreadHandler. As you mentioned earlier in one of my PRs, we need to explicitly find the the cause that raises the error hence we are doing everything except passing values of text and updated_status